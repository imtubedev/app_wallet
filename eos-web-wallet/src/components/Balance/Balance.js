import * as React from "react";
import numeral from "numeral";
import css from "./styles.module.scss";
import cx from "classnames";
import ScaleText from "react-scale-text";

const Balance = ({ total, difference }) => {
  let totalBalance = numeral(total).format("0.00a");
  let balanceSuffix;
  if (totalBalance.slice(-1).match(/[a-zA-Z]/)) {
    balanceSuffix = totalBalance.slice(-1);
    totalBalance = totalBalance.slice(0, totalBalance.length - 1);
  }

  return (
    <div className="-is-logged-in">
      <div className={cx("financials", css.financials)}>
        <ScaleText widthOnly>
          <p className={cx(css.balance)}>
            {totalBalance}
            {balanceSuffix && (
              <span className={css.balance_suffix}>{balanceSuffix}</span>
            )}
          </p>
        </ScaleText>
        <p className={cx("full_balance is-6", css.full_balance)}>
          {numeral(total).format("0,0.0000")}
        </p>
        {!!difference && (
          <div
            className={cx("tag change", css.tag, {
              [css.minus]: difference < 0
            })}
          >
            {difference > 0 && "+"}
            {difference.toFixed(4)}
          </div>
        )}
      </div>
    </div>
  );
};

export default Balance;
