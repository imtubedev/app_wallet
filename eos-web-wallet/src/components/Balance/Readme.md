A Balance example with positive difference:

```js
<div style={{ backgroundColor: "#575f68", padding: "2rem"}}>
  <Balance total="200" difference={0.0001} symbol={<span className="icon-increase"></span>} />
</div>
```

A Balance example with negative difference:

```js
<div style={{ backgroundColor: "#575f68", padding: "2rem"}}>
  <Balance total="200" difference={0.0001} symbol={<span className="icon-decrease"></span>} />
</div>
```

A Balance example with nothing:

```js
<div style={{ backgroundColor: "#575f68", padding: "2rem"}}>
  <Balance total="0" />
</div>
```

A Balance example with many decimals:

```js
<div style={{ backgroundColor: "#575f68", padding: "2rem"}}>
  <Balance total="200.23223232" difference={0.232323} symbol={<span className="icon-increase"></span>} />
</div>
```

A large total value Balance example with negative difference:

```js
<div style={{ backgroundColor: "#575f68", padding: "2rem"}}>
  <Balance total="20000000.23223232" difference={-0.232323} symbol={<span className="icon-increase"></span>} />
</div>
```
