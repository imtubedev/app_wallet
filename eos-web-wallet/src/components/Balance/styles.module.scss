@import "../../styles/custom_variables.scss";

.financials {
  .balance {
    line-height: normal;
    font-weight: 200;
    min-width: 200px;
  }
  .full_balance {
    font-size: 1.125rem;
    line-height: 1rem;
    margin: 0;
    margin-bottom: 0 !important;
    color: $grey-text;
    font-weight: 400;
  }
  .tag {
    font-weight: bold;
    font-size: 0.875rem;
    background-color: $green;
    color: $grey-light;
    margin-top: 5px;
    span {
      font-size: 0.5625rem;
      padding-right: 2px;
    }
    &.minus {
      background-color: $red;
      color: $white;
    }
  }
  .balance_suffix {
    font-size: 50%;
    text-transform: uppercase;
  }
}
