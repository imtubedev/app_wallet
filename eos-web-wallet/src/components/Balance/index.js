import Balance from "./Balance";
import {
  withCurrentEOSBalance,
  withEOSBalanceForAccount
} from "../../containers/balance";

export const UserEOSBalance = withEOSBalanceForAccount(Balance);

export const CurrentEOSBalance = withCurrentEOSBalance(Balance);

export default Balance;
