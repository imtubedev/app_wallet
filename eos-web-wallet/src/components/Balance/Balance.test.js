import * as React from "react";
import renderer from "react-test-renderer";
import Balance from "./Balance";

describe("<Balance />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<Balance />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
