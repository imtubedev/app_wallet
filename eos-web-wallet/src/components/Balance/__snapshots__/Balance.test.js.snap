// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Balance /> renders without crashing 1`] = `
<div
  className="-is-logged-in"
>
  <div
    className="financials"
  >
    <div
      className="scaletext-wrapper"
      style={
        Object {
          "fontSize": "inherit",
          "height": "auto",
          "overflowX": "hidden",
          "overflowY": "visible",
          "width": "100%",
        }
      }
    >
      <p
        className=""
        fontSize="inherit"
      >
        0.00
      </p>
    </div>
    <p
      className="full_balance is-6"
    >
      0.0000
    </p>
  </div>
</div>
`;
